src/Chainweb/Pact/PactService.hs

```diff
@@ -538,9 +549,13 @@ execNewGenesisBlock
 execNewGenesisBlock miner newTrans = withDiscardedBatch $ do
     pdbenv <- restoreCheckpointer Nothing
     results <- execTransactions Nothing miner newTrans pdbenv
+    finalizeCheckpointer discard
     return $! toPayloadWithOutputs miner results
 
-
+-- |
+-- Precondition: restoreCheckpointer
```

Comment:
```suggestion
```

`restoreCheckpointer` is not a precondition but called within the function in the call to `rewindTo`.